qiskit/transpiler/passes/layout/vf2_layout.py

```diff
@@ -218,7 +218,7 @@ def mapping_to_layout(layout_mapping):
                 )
                 chosen_layout = layout
                 chosen_layout_score = layout_score
-            if self.max_trials is not None and self.max_trials > 0 and trials >= self.max_trials:
+            if self.max_trials and trials >= self.max_trials:
```

Comment:
This is a bit of an unrelated change, it looks completely correct to me but in the interest of keeping this PR isolated to a single logical change do you mind reverting this. We can fix it as part of a follow up PR. (since this is a feature freeze exception I'm kind of keen to keep this isolated to just the minimum piece needed for setting a trial limit in vf2postlayout)